File: 프로그래머스/3/43238. 입국심사/입국심사.py
def solution(n, times):
    left = 0
    right = min(times) * n
    while left <= right:
        cnt = 0
        mid = (left + right) // 2
        for time in times:
            cnt += mid // time
        
        if cnt >= n:
            right = mid-1
        else:
            left = mid+1
    
    return left

File: 프로그래머스/3/43238. 입국심사/입국심사.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

long long solution(int n, vector<int> times) {
    // long long answer = 0;
    long long left = 0;
    long long right = (long long)*min_element(times.begin(), times.end()) * n;
    
    while (left < right){
        long long mid = (left + right) / 2;
        long long cnt = 0;
        for (auto t : times){
            cnt += mid / t;
            if (cnt >= n) break;
        }
        if (cnt >= n){
            right = mid;
        }
        else{
            left = mid+1;
        }
    }
    
    return left;
}